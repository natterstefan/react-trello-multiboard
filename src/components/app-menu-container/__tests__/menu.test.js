import React from 'react'
import { mount, shallow } from 'enzyme'
import { HashRouter as Router } from 'react-router-dom'
import ListItem from '@material-ui/core/ListItem'

import AppMenuContent from '../menu'

describe('Component/AppMenuContent', () => {
  test('should render without throwing an error', () => {
    expect(shallow(<AppMenuContent />)).toMatchSnapshot()
  })

  test('should pass the correct properties to the rendered ListItems', () => {
    const wrapper = mount(
      <Router>
        <AppMenuContent />
      </Router>,
    )
    expect(wrapper.find(ListItem).length).toBe(3)
    expect(
      wrapper
        .find(ListItem)
        .first()
        .prop('to'),
    ).toBe('/')
    expect(
      wrapper
        .find(ListItem)
        .at(1)
        .prop('to'),
    ).toBe('/config')
  })
})
