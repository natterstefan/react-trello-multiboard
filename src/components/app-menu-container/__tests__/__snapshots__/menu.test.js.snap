// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/AppMenuContent should render without throwing an error 1`] = `
<div>
  <WithStyles(List)
    component="nav"
  >
    <WithStyles(ListItem)
      button={true}
      component={[Function]}
      replace={true}
      to="/"
    >
      <WithStyles(ListItemIcon)>
        <pure(Dashboard) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        primary="Dashboard"
      />
    </WithStyles(ListItem)>
    <WithStyles(ListItem)
      button={true}
      component={[Function]}
      replace={true}
      to="/config"
    >
      <WithStyles(ListItemIcon)>
        <pure(Settings) />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        primary="Config"
      />
    </WithStyles(ListItem)>
    <WithStyles(ListItem)
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ListItemIcon)>
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "github",
            ]
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          symbol={false}
          transform={null}
        />
      </WithStyles(ListItemIcon)>
      <WithStyles(ListItemText)
        primary="GitHub"
      />
    </WithStyles(ListItem)>
  </WithStyles(List)>
</div>
`;
